src/Extensions/Features/test/FeatureCollectionTests.cs

```diff
@@ -44,4 +45,61 @@ public void SetNullValueRemoves()
         var thing2 = interfaces[typeof(IThing)];
         Assert.Null(thing2);
     }
+
+    [Fact]
+    public void GetMissingStructFeatureThrows()
+    {
+        var interfaces = new FeatureCollection();
+
+        // Regression test: Used to throw NullReferenceException because it tried to unbox a null object to a struct
+        Assert.Throws<InvalidOperationException>(() => interfaces.Get<int>());
+    }
+
+    [Fact]
+    public void GetMissingFeatureReturnsNull()
+    {
+        var interfaces = new FeatureCollection();
+
+        Assert.Null(interfaces.Get<Thing>());
+    }
+
+    [Fact]
+    public void GetStructFeature()
+    {
+        var interfaces = new FeatureCollection();
+        var value = 20;
+        interfaces.Set(value);
+
+        Assert.Equal(value, interfaces.Get<int>());
+    }
+
+    [Fact]
+    public void GetNullableStructFeatureWhenSetWithNonNullableStruct()
+    {
+        var interfaces = new FeatureCollection();
+        var value = 20;
+        interfaces.Set(value);
+
+        Assert.Null(interfaces.Get<int?>());
```

Comment:
Feels a little weird that `int` and `int?` are stored separately, but I assume it doesn't matter in practice.

Reply:
The key is the type in the feature collection.

We could unwrap `Nullable<T>` on get/set, but no one has asked for it, and it feels like something that is a little too clever that will cause unanticipated problems later.